### ISSUE TITLE
Linux / MacOS Support

### ISSUE THREAD

**Issue by osheari1** [BenediktAlkin/SongTaggerForSpotify#18]
This is an amazing idea, so excellent work! However only packaging for Windows is painful for linux / mac users. Is there anyway this could at least be easily dockerized? Packaging for other systems directly would also be awesome. I'm not a Mono expert, but maybe it's as simple as changing the compilation target?

**Comment by BenediktAlkin** [comment]
The reason for Windows only is mainly because I'm not an UI expert and was familiar with WPF. I looked into some things how one could run WPF apps in linux/mac but I didn't manage to get it running with anything. 

The main reason that it doesn't work is because WPF requires the native Windows UI components and therefore it's not cross-platform, even though all other things of the app would be cross-platform.

I am planning to make the app cross-platform, but this takes a lot of effort and as I'm currently busy with work/other stuff I don't have the time for it.

**Comment by osheari1** [comment]
Ok cool understood. Front end and I don't play well together so I feel ya. Is there by chance an easy way to run it in server only mode? At least from there we can make use of the API.

**Comment by BenediktAlkin** [comment]
Yes, there is an [API](https://github.com/BenediktAlkin/SongTaggerForSpotify/blob/main/BackendAPI/documentation.md) but it doesn't support all operations (mainly used for integration with [Tagify](https://github.com/vokinpirks/Tagify)

**Comment by CavalierBen** [comment]
While more native support is always nice, just wanted to chime in that I was surprised at how well wine (7.19-1) worked my first try run- running `wine SpotifySongTagger.exe` prompted downloading and installing the .NET runtime also in wine- so far I've been able to tag songs and create playlists without any notable issues, so definitely worth trying first :smile:

**Comment by Chipwingg** [comment]
I have this working beautifully under Linux, using `wine-8.1 (Staging)` and `.NET 5.0.17` downloaded from Microsoft. 

As for macOS, I am not a Mac+Wine expert, and I don't even know where to begin troubleshooting. Have there been any reports of this running on macOS, or am I on my own?

**Comment by echennh** [comment]
Same here, thanks for making this program, it's exactly what I was looking for! 
My laptop is also Linux, so I've downloaded wine and .NET as per @Chipwingg 's suggestion, and the program seems to be successfully installed and I successfully logged in, but ultimately the app crashed with an error. I'm including the [backtrace.txt](https://github.com/user-attachments/files/16603061/backtrace.txt) here. 
So in short, it looks like I'm gonna have to wait until @BenediktAlkin has time to update this app to support Linux 🙏🏼
